snuba/reader.py

```diff
@@ -96,3 +98,30 @@ def execute(
             ),
             with_totals=with_totals,
         )
+
+
+class HTTPReader(Reader):
+    def __init__(
+        self, host: str, port: int, options: Optional[Mapping[str, str]] = None
+    ):
+        self.__base_url = f"http://{host}:{port}/"
+        self.__options = options if options is not None else {}
+
+    def execute(
+        self,
+        query: str,
+        settings: Optional[Mapping[str, str]] = None,
+        query_id: Optional[str] = None,
+        with_totals: bool = False,
+    ) -> Result:
+        parameters = {**self.__options}
+        if settings is not None:
+            parameters.update(settings)
+        if query_id is not None:
+            parameters["query_id"] = query_id
+        response = requests.post(
+            urljoin(self.__base_url, "?" + urlencode(parameters)),
+            data=query.encode("utf-8"),
+        )
```

Comment:
This could probably use straight `urllib3`, but I used `requests` here for consistency with the [`HTTPBatchWriter`](https://github.com/getsentry/snuba/blob/1e4e383fa00ea82d3dfcbeb7ba7fa7f1923d2aee/snuba/writer.py#L62-L64).

Reply:
Not anymore @tkaemming.

Reply:
💀

Reply:
Used 97b36c7e5592a2a020aa548364184eaa731d4b8e as the pattern for this change. Probably need to take a closer look at request headers.